desktop/libs/indexer/src/indexer/indexers/sql.py

```diff
@@ -292,13 +294,20 @@ def create_table_from_local_file(self, source, destination, csv_data, start_time
           'primary_keys': ', '.join(destination.get('indexerPrimaryKey'))
       }
 
-    for csv_row in csv_data:
-      sql += '''
-          INSERT INTO %(table_name)s VALUES %(csv_row)s;
-          ''' % {
-                  'table_name': table_name,
-                  'csv_row': tuple(csv_row)
-                }
+    path = source['path']
+    if path:
+      with open(BASE_DIR + path, 'r') as local_file:
+        reader = csv.reader(local_file)
+        list_of_tuples = list(map(tuple, reader))
+
+        if source['format']['hasHeader']:
+          list_of_tuples = list_of_tuples[1:]
+
+        csv_rows = str(list_of_tuples)[1:-1]
+        sql += '''INSERT INTO %(table_name)s VALUES %(csv_rows)s;'''% {
+            'table_name': table_name,
+            'csv_rows': csv_rows
+          }
```

Comment:
And this execution is very fast as compared to for loop. We are inserting all values in a single query.

Reply:
Big +1

(Phoenix did not support bulk INSERTs, glad we can do it now with Hive/Impala!)